core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -1706,6 +1758,279 @@ public RemoteLogMetadataManager createRemoteLogMetadataManager() {
         }
     }
 
+    @Test
+    void testCorruptedTimeIndex() throws Exception {
```

Comment:
There is lot of duplicated code amongst the three tests added here. Can we use `@ParameterizedTest` and combine them together?